### PR TITLE
Adding separate extension_directory configuration setting

### PR DESCRIPTION
**Pull request by hannes**
DuckDB extensions are by default installed in a `.duckdb` subfolder of whatever the user's home directory is. In this PR we add a separate setting `extension_directory` that can override where those extensions are going to be installed and looked for. This is required for CRAN compliance as for R extensions are not allowed to write elsewhere on the file system.

### CODE REVIEW

File: tools/rpkg/R/dbFetch__duckdb_result.R [hunk 1]
```diff
@@ -35,10 +35,10 @@ dbFetch__duckdb_result <- function(res, n = -1, ...) {
...
     stop("n needs to be not a whole number")
   }
-  if (res@stmt_lst$type != "SELECT" && res@stmt_lst$type != "RELATION" && res@stmt_lst$return_type != "QUERY_RESULT") {
-    warning("Should not call dbFetch() on results that do not come from SELECT, got ", res@stmt_lst$type)
-    return(data.frame())
-  }
+#   if (res@stmt_lst$type != "SELECT" && res@stmt_lst$type != "RELATION" && res@stmt_lst$return_type != "QUERY_RESULT") {
+#     warning("Should not call dbFetch() on results that do not come from SELECT, got ", res@stmt_lst$type)
+#     return(data.frame())
+#   }
```
[the remark this answers is not in the document]
Comment (krlmlr) [reply tools/rpkg/R/dbFetch__duckdb_result.R:41]: I rather like the distinction between a query that returns a tabular result set, and a statement that is called for its side effects. Should we discuss this in a separate PR/issue?

[the remark this answers is not in the document]
Comment (hannes) [reply tools/rpkg/R/dbFetch__duckdb_result.R:41]: Thing is there are so many queries that return result sets, e.g. INSERT with RETURNING clauses etc.

[the remark this answers is not in the document]
Comment (krlmlr) [reply tools/rpkg/R/dbFetch__duckdb_result.R:41]: I remember a better way in duckdb to decide if a statement returns a table. I can take a closer look.

[the remark this answers is not in the document]
Comment (hannes) [reply tools/rpkg/R/dbFetch__duckdb_result.R]: I've removed the change again, its a different PR

File: tools/rpkg/R/Driver.R [hunk 2]
```diff
@@ -40,6 +40,11 @@ duckdb <- function(dbdir = DBDIR_MEMORY, read_only = FALSE, bigint = "numeric",
     stop(paste("Unsupported bigint configuration", bigint))
   )
 
+  # R packages are not allowed to write extensions into home directory, so use R_user_dir instead
+  if (!("extension_directory" %in% names(config))) {
+    config["extension_directory"] = tools::R_user_dir("duckdb", "data")
```
Comment (krlmlr) [inline tools/rpkg/R/Driver.R:45]: Perhaps:

```suggestion
    config[["extension_directory"]] <- tools::R_user_dir("duckdb", "data")
```

File: tools/rpkg/R/sql.R [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#' `sql()` runs an arbitrary SQL query and returns a data.frame the query results
+#'
+#' @param sql A SQL string
+#' @param conn An optional connection, defaults to built-in default
+#' @return A data frame with the query result
+#' @export
+#' @examples
+#' print(sql("SELECT 42"))
+
+sql <- function(sql, conn = default_connection()) {
```
Comment (krlmlr) [inline tools/rpkg/R/sql.R:12]: This name will clash with dplyr/dbplyr.

Perhaps `duckdb_sql()` ?

Comment (hannes) [reply tools/rpkg/R/sql.R:12]: I'd rather have the shorthand in duckdb than not clash tbh

File: tools/rpkg/R/Driver.R (hunk 2 above)
Comment (hannes) [reply tools/rpkg/R/Driver.R:45]: Good call!

**Comment by hannes** [comment]
Not sure about that UBSAN issue, seems unrelated

**Comment by hannes** [comment]
Maybe dplyr should stop exporting `sql`...

**Pull request merged**